File: tradepulse_ai_package/frontend/src/pages/signals.js
import Layout from '../components/Layout';
import SignalCard from '../components/SignalCard';
const signals = ['BTC/USDT','ETH/USDT','EUR/USD','XAU/USD','GBP/USD'].map((s,i)=>({symbol:s,direction:i%2?'SELL':'BUY',confidence:68+i*3,entry:'Market zone',stop_loss:'Auto volatility SL',take_profit:'Auto risk/reward TP',risk_reward:'1:2',confirmations:[{name:'RSI',status:'checked'},{name:'MACD',status:'checked'},{name:'MA',status:'checked'},{name:'News',status:'checked'}]}));
export default function Signals(){ return <Layout><h1>Live Signals</h1>{signals.map(s=><SignalCard key={s.symbol} signal={s}/>)}</Layout> }

File: tradepulse_ai_package/backend/app/routes/signals.py
from fastapi import APIRouter
from app.services.signal_engine import generate_signal

router = APIRouter()

@router.get("/")
def get_signals():
    symbols = ["BTC/USDT", "ETH/USDT", "EUR/USD", "XAU/USD", "GBP/USD"]
    return [generate_signal(symbol) for symbol in symbols]

@router.get("/{symbol}")
def get_signal(symbol: str):
    return generate_signal(symbol.upper())
